docs/pages/error_handling.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: default
+title: Error Handling
+description: "Error handling"
+permalink: /error-handling
+---
+
+# Error Handling
+{: .no_toc}
+
+Connectors can throw a variety of different error types.
+The type of error thrown has an effect on workflow behaviour and how the error is handled.
+
+The `@trayio/connector-utils` node package has some helper functions included. 
+You can install the package by running `npm i --save @trayio/connector-utils`.
+
+## Error Types
+{: .no_toc}
+
+- TOC
+{:toc}
+
+### User Input Error
+
+A user input error should be thrown when you are sure the user input is wrong and the API call will never be valid.
+When a user input is thrown, the step **will not retry**.
+
+```js
+throw new Error({
+  code: '#user_input_error',
+  message: 'Please supply at least one of Status Text, Status Emoji, or Status Expiration.',
+});
+```
+
+_With connector-utils_
+
+```js
+const { UserInputError } = require('@trayio/connector-utils').error;
+
+throw new UserInputError('This address entered is invalid');
+```
+
+### API Error
+
+An API error should be thrown if the error is caused by the API.
+Steps that fail with an API Error will retry automatically.
+
+If the connector is configured with `expects` to check the response status code,
+this is the type of error that is thrown when the status code is out of the accepted range.
+
+```js
+throw new Error({
+  code: '#api_error',
+  message: 'API returned a status code of 404',
+});
+```
+
+_With connector-utils_
+
+```js
+const { ApiError } = require('@trayio/connector-utils').error;
+
+throw new ApiError('Login to the API  failed.');
+```
+
+### Connector Error
+
+This is the default error type for when there is a code problem.
+If your code throws an exception that isn't caught, it will default to a Connector Error.
+
+Connector Errors are not displayed to the user, as they will just see a message like `Unfortunately, the connector unexpectedly failed`.
+When this error type is thrown, the step will retry by default.
+
+_With connector-utils_
+
+```js
+const { ConnectorError } = require('@trayio/connector-utils').error;
+
+throw new ConnectorError('Failed to parse downloaded JSON');
+```
+
+### OAuth Refresh Error
+
+OAuth Refresh errors are used signal that the access token has expired and needs refreshing. 
+If your connetors uses OAuth2, it is recommended that you add checks in `afterFailure` to check for the appropriate status code and throw this error when needed.
```

Comment:
A couple of typos. Should be:
OAuth Refresh errors are used to signal that the access token has expired and needs refreshing. 
If your connector uses OAuth2, it is recommended that you add checks in `afterFailure` to check for the appropriate status code and throw this error when needed.